### ISSUE TITLE
DeiT : Training data-efficient image transformers & distillation through attention

### ISSUE THREAD

**Issue by yskim0** [One-Day-One-Paper/One-Day-One-Paper#15]
# Abstract

- ViT → 매우 많은 데이터를 필요로 함
    
    ⇒ Motivation : 적은 데이터로 좋은 성능 내자
    
- training on **Imagenet only** → ViT에 준하는 performance 성취
    
    ![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/d516e59a-0411-4fc7-b6e9-f0524c8d87d9/Untitled.png)
    
    acc. & speed trade-off
    
- **introduce a teacher-student strategy to transformers**
- add a **distillation token** ensuring that the student learns from the teacher through attention

# Introduction

- knowledge distillation을 사용한다면 "how to distll these models?"
    - → **token-based strategy**
- distillation token
    - same role as the class token, except that it aims at re-producing the label estimated by the teacher
    - this strategy outperforms vanilla distillation
- summary
    - no conv. layer, no external data
    - propose a new distillation procedure based on a distillation token
    - our distillation strategy → image transformers learn more from a convnet than from another transformer with comparable performance
        - **teacher model : CNN based model > Transformer**
            - **∵ inductive bias**

# Related Works

- Knowledge Distillation(KD)
    - training paradigm in which a *student* model leverages “soft” labels coming
    from a strong *teacher* network.
    - output vector of the teacher’s softmax function ⇒ a “hard” label.
    - student model → it can be regarded as a form of compression of the teacher model into a smaller one
    - the teacher’s supervision takes into account the effects of the data augmentation
    - in this paper, student → transformer
        
        teacher → convnet or transformer 
        

# Methods

💥 assume that we have access to a strong image classifier as a teacher model (convnet, or a mixture of classifiers are possible)

## Distillation Loss — Soft vs. Hard

- Soft distillation
    
    ![스크린샷 2021-11-10 오전 2.14.16.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/cb0b8add-d0d8-4fe1-bb60-7ce4334d21a8/스크린샷_2021-11-10_오전_2.14.16.png)
    
    - min. the KL divergence bet. the softmax of the tacher and the softmax of the student model
- Hard-label distillation
    
    ![스크린샷 2021-11-10 오전 2.15.18.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/48593646-b7ab-4dc1-a03f-6b2a14517aa4/스크린샷_2021-11-10_오전_2.15.18.png)
    
    - hard decision of the teacher as a true label
    - $y_t = argmax_cZ_t(c)$
    - the hard label associated with the teacher may change depending on the specific data augmentation
    - note also that the hard lables can also be converted into soft labels with label smoothing~

## Distillation Token

![스크린샷 2021-11-10 오전 2.16.23.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/ae40fa03-67cd-455c-915d-1cc28a2de33c/스크린샷_2021-11-10_오전_2.16.23.png)

- Distillation token
    - add a new token, the distillation token, to the initial embeddings(patches and class tokens)
    - it interacts with other embedding through self-attention
    - The distillation embedding allows our model to learn from the output of the teacher
- the learned class and distillation tokens converge towards different vectors
    - the average cosine similarity between these tokens equal to 0.06
- the class and distillation embeddings are computed at each layer, they gradually become more similar through the network, all the way through the last layer at which their similarity is high (cos=0.93)
- distillation token 안쓰고 class token 그대로 다시 사용하면?
    - they converge towards the same vector (cos=0.999), 
    the output embedding are also quasi-identical.
    - This additional class token does not bring anything to the classification performance

## Joint classifiers

class embedding vector, distillation embedding vector를 test 시 어떻게 사용하는가

1. concat → FC → predict
2. late fusion fashion
    - add the softmax output by the two classifiers to make the prediction
